reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/TimedScheduler.java

```diff
@@ -0,0 +1,274 @@
+
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted for immediate execution
+	 * (ie. {@link Scheduler#schedule(Runnable)} or {@link Worker#schedule(Runnable)}).
+	 */
+	static final String SUBMISSION_DIRECT = "direct";
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted with a delay
+	 * (ie. {@link Scheduler#schedule(Runnable, long, TimeUnit)}
+	 * or {@link Worker#schedule(Runnable, long, TimeUnit)}).
+	 */
+	static final String SUBMISSION_DELAYED = "delayed";
+	/**
+	 * {@link Counter} that increments when a task is initially submitted with a period
+	 * (ie. {@link Scheduler#schedulePeriodically(Runnable, long, long, TimeUnit)}
+	 * or {@link Worker#schedulePeriodically(Runnable, long, long, TimeUnit)}). This isn't
+	 * incremented on each re-run of the periodical task.
+	 */
+	static final String SUBMISSION_PERIODICALLY_INITIAL = "periodically_initial";
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted with a period
+	 * (ie. {@link Scheduler#schedulePeriodically(Runnable, long, long, TimeUnit)}
+	 * or {@link Worker#schedulePeriodically(Runnable, long, long, TimeUnit)}), both initially AND
+	 * each time the task is re-run.
+	 */
+	static final String SUBMISSION_PERIODICALLY_ITERATION = "periodically_iteration";
+
+	final Scheduler delegate;
+
+	final MeterRegistry registry;
+
+	final Counter       submittedDirect;
+	final Counter       submittedDelayed;
+	final Counter       submittedPeriodicallyInitial;
+	final Counter       submittedPeriodicallyIteration;
+	final LongTaskTimer pendingTasks;
+	final LongTaskTimer activeTasks;
+	final Timer         completedTasks;
+
+
+	TimedScheduler(Scheduler delegate, MeterRegistry registry,  String metricPrefix, Iterable<Tag> tagsList) {
+		this.registry = registry;
+		this.delegate = delegate;
+		if (!metricPrefix.endsWith(".")) {
+			metricPrefix = metricPrefix + ".";
+		}
+
+		Tags tags = Tags.of(tagsList);
+
+		this.submittedDirect = registry.counter(metricPrefix + METER_SUBMITTED, tags.and(TAG_SUBMISSION, SUBMISSION_DIRECT));
+		this.submittedDelayed = registry.counter(metricPrefix + METER_SUBMITTED, tags.and(TAG_SUBMISSION, SUBMISSION_DELAYED));
+		this.submittedPeriodicallyInitial = registry.counter(metricPrefix + METER_SUBMITTED, tags.and(TAG_SUBMISSION, SUBMISSION_PERIODICALLY_INITIAL));
+		this.submittedPeriodicallyIteration = registry.counter(metricPrefix + METER_SUBMITTED, tags.and(TAG_SUBMISSION, SUBMISSION_PERIODICALLY_ITERATION));
+
+		this.pendingTasks = LongTaskTimer.builder(metricPrefix + METER_TASKS_PENDING)
+			.tags(tags).register(registry);
+		this.activeTasks = LongTaskTimer.builder(metricPrefix + METER_TASKS_ACTIVE)
+			.tags(tags).register(registry);
+		this.completedTasks = registry.timer(metricPrefix + METER_TASKS_COMPLETED, tags);
+
+	}
+
+	Runnable wrap(Runnable task) {
+		return new TimedRunnable(registry, this, task);
+	}
+
+	Runnable wrapPeriodical(Runnable task) {
```

Comment:
I suppose `wrapPeriodic` would be more accurate (https://grammarist.com/usage/periodic-periodical/)

Reply:
interesting, thanks for pointing that out ! I have further changed the wording to `periodic` in all places (including variable names, constant names, constant strings and javadocs/comments)

I also fixed it in test names.